### PR TITLE
[THOR-842] - Added new option for a sort field and sort direction

### PR DESCRIPTION
**Pull request by vegaed**
* Seperated showing the date from sorting
* Added two new option to support

### CODE REVIEW

**Comment by rartiss55** [comment]
Shouldn't this be merged into lorelei-master first and not master?

**Comment by ThomasSchellenbergNextCentury** [comment]
Rachel - Fixed.

**Comment by ThomasSchellenbergNextCentury** [comment]
Sorry - Changed the base and then changed back to update the diff.

**Comment by ThomasSchellenbergNextCentury** [comment]
@vegaed ~~Please add sortField to updateFieldsOnTableChanged in the DocumentViewerOptions (in document-viewer.component.ts)~~

Actually, nevermind.  I forgot that the changes to the Options (and the removal of updateFieldsOnTableChanged) are already merged-into lorelei-master.  I'll handle this during merge conflict resolution.

**Comment by rartiss55** [comment]
@vegaed Please fix/update the unit tests for document viewer -- I'm getting a failure for this test, probably because of the added sort by clause: "returns expected query from createQuery"

**Comment by vegaed** [comment]
@rartiss55 good catch. All fixed.

File: src/app/components/document-viewer/document-viewer.component.spec.ts [hunk 1]
```diff
@@ -170,22 +170,39 @@ describe('Component: DocumentViewer', () => {
...
-                'testTextField',
-                'testIdField'
-            ])
+            .withFields(['testTextField', 'testIdField'])
             .limit(50)
             .offset(0);
         expect(component.createQuery()).toEqual(query);
 
-        // Then add a date field and ensure the result is properly sorting.
+        // Then add a date field and ensure the sort is not effectivedthe result is properly sorting.
```
Comment (rartiss55) [inline src/app/components/document-viewer/document-viewer.component.spec.ts]: Did this comment get changed by accident?

File: src/app/components/document-viewer/document-viewer.component.spec.ts [hunk 2]
```diff
@@ -170,22 +170,39 @@ describe('Component: DocumentViewer', () => {
...
+        expect(component.createQuery()).toEqual(query);
+    });
+
+    it('returns expected query from createQuery with sort', () => {
+        component.options.database = new DatabaseMetaData('testDatabase1');
+        component.options.table = new TableMetaData('testTable1');
+        component.options.dataField = DatasetServiceMock.TEXT_FIELD;
+        component.options.dateField = DatasetServiceMock.DATE_FIELD;
+        component.options.idField = DatasetServiceMock.ID_FIELD;
+        component.options.sortField = DatasetServiceMock.NAME_FIELD;
```
Comment (rartiss55) [inline src/app/components/document-viewer/document-viewer.component.spec.ts]: Looking over this again, is there actually a way to set the sortField in the config? I was able to update it in the app via settings but setting in directly in config.json didn't seem to work. 

Also it should probably be added to Document Viewer section here:
https://github.com/NextCenturyCorporation/neon-dash/wiki/Neon-Dashboard-Configuration-Guide

Comment (wagnerwj) [reply src/app/components/document-viewer/document-viewer.component.spec.ts]: The code references a property in the options:
  if (this.options.sortField) {
            query = query.sortBy(this.options.sortField.columnName,
                (this.options.sortOrder === 'DESCENDING') ? neonVariables.DESCENDING : neonVariables.ASCENDING);
}
however, this element doesn't appear to be in BaseNeonOptions on that branch.

Comment (ThomasSchellenbergNextCentury) [reply src/app/components/document-viewer/document-viewer.component.spec.ts]: @rartiss55 There was an issue caused by the fact that Edgardo branched off an older version of lorelei-master.  The fields are read from the config file differently in the new version.  I've merged the latest into this branch and updated the document viewer so the sort field should be read correctly now.  Could you please confirm that it works for you?  Thanks!

File: src/app/components/document-viewer/document-viewer.component.spec.ts (hunk 1 above)
Comment (ThomasSchellenbergNextCentury) [reply src/app/components/document-viewer/document-viewer.component.spec.ts]: Fixed

File: src/app/components/document-viewer/document-viewer.component.spec.ts (hunk 2 above)
Comment (rartiss55) [reply src/app/components/document-viewer/document-viewer.component.spec.ts]: @ThomasSchellenbergNextCentury -- Thanks! Looks like it works now. I'll go ahead and approve.

**Review by rartiss55 — APPROVED** [review]

**Review by ThomasSchellenbergNextCentury — APPROVED** [review]

**Pull request merged**